### ISSUE TITLE
bad autoscale_view() with long curved FancyArrowPatch

### ISSUE THREAD

**Issue by grahamgower** [matplotlib/matplotlib#19723]
### Bug report

When using a very long `FancyArrowPatch` with `connectionstyle="arc3"`, subsequent axes autoscaling can badly squash the figure's content. Example below exhibits the problem, but more extreme squashing can easily be obtained. Possibly this is not limited to `FancyArrowPatch`?

**Code for reproduction**

```python
import matplotlib
import matplotlib.pyplot as plt

fap = matplotlib.patches.FancyArrowPatch(
    (10, 10),
    (20, 2e4),
    mutation_scale=50,
    connectionstyle="arc3,rad=0.3",
)

_, ax = plt.subplots()
ax.add_patch(fap)
ax.set_title(f"matplotlib {matplotlib.__version__}")
ax.autoscale_view()
ax.figure.savefig("curved-arrow.png")
```

**Actual outcome**

![curved-arrow](https://user-images.githubusercontent.com/8665077/111459663-45cf1c80-871b-11eb-9255-cf475ef1725b.png)

**Expected outcome**

The axes view should span the drawn content. Any control point(s) required for the arc should be ignored.

**Matplotlib version**

  * Operating system: Linux
  * Matplotlib version: 3.3.4
  * Matplotlib backend: Reproducible with at least `GTK3Agg` and `agg`.
  * Python version: 3.9

Matplotlib installed via pip.

**Comment by QuLogic** [comment]
This is a duplicate of #19174, I believe.

**Comment by dstansby** [comment]
I don't think it is; https://github.com/matplotlib/matplotlib/pull/19214 seems to fix https://github.com/matplotlib/matplotlib/issues/19174 but not this.

**Comment by dstansby** [comment]
I can still reproduce this on current main.

**Comment by oscargus** [comment]
This is probably the source (which may be obvious, but still):
```
>>> fap.get_extents()
Bbox([[6.742420563716048, 9.1385591605688], [3018.5610383181834, 19998.286260206016]])
```
